monitoring/harmony-monitor/blockchain-watchdog/reporting-server.go

```diff
@@ -202,21 +200,31 @@ func (m *monitor) renderReport(w http.ResponseWriter, req *http.Request) {
 			sum[headerSumry][i].(any)["shard-leader"] = leaders[i]
 		}
 	}
+	m.use()
 	cnsMsg := "Consensus Progress not known yet"
 	if len(m.consensusProgress) != 0 {
 		cM, _ := json.Marshal(m.consensusProgress)
 		cnsMsg = fmt.Sprintf("Consensus Progress: %s", cM)
 	}
+	totalNoReplyMachines := []noReply{}
+	totalNoReplyMachines = append(totalNoReplyMachines, m.MetadataSnapshot.Down...)
+	totalNoReplyMachines = append(totalNoReplyMachines, m.BlockHeaderSnapshot.Down...)
+	m.done()
 	t.ExecuteTemplate(w, "report", v{
 		LeftTitle:  []interface{}{m.chain, cnsMsg},
 		RightTitle: []interface{}{buildVersion, time.Now().Format(time.RFC3339)},
 		Summary:    sum,
-		NoReply:    m.NoReplyMachines,
-		DownMachineCount: linq.From(m.NoReplyMachines).Select(
+		NoReply:    totalNoReplyMachines,
+		DownMachineCount: linq.From(totalNoReplyMachines).Select(
 			func(c interface{}) interface{} { return c.(noReply).IP },
 		).Distinct().Count(),
 	})
-	m.SummarySnapshot = sum
+	m.use()
+	m.SummarySnapshot = make(map[string]map[string]interface{})
+	copier.Copy(&m.SummarySnapshot, &sum)
```

Comment:
I don't want to use this library, it relies on reflection https://github.com/jinzhu/copier/blob/master/copier.go#L14 , that is very slow. Hit to performance not worth the convenience